chatterbot/algorithms/engram.py

```diff
@@ -1,43 +1,38 @@
-def get_closest(text, log_directory, similarity_threshold):
-    from chatterbot.conversation import Statement, Conversation
-    import os
-
-    closest_response = []
-    closest_ratio = 0
+def engram(chatbot, text):
+    """
+    Takes a chatbot object and a statement object.
+    Returns a response based on the closest match based on in known conversations.
+    """
+    from bson import ObjectId
+    from fuzzywuzzy import fuzz
 
-    for log in os.listdir(log_directory):
-        path = log_directory + "/" + log
+    statements = chatbot.database.statements
 
-        if os.path.isfile(path):
-            conversation = Conversation()
-            conversation.read(path)
-            response, ratio = conversation.find_closest_response(text)
+    closest_statement = statements.find()[0]
+    closest_ratio = 0
 
-            # A response may not be returned if a log has less than one line
-            if response:
-                if ratio > closest_ratio:
-                    closest_response = []
-                    closest_response.append(response)
-                    closest_ratio = ratio
-                elif ratio == closest_ratio and closest_ratio != 0:
-                    closest_response.append(response)
-                    closest_ratio = ratio
+    for statement in statements.find():
+        ratio = fuzz.ratio(statement["text"], text)
 
-    return closest_response, closest_ratio
+        responces_exist = statements.find(
+            {"in_response_to": statement["_id"]}
+        ).count() > 0
 
-def engram(text, log_directory):
-    """
-    Takes a message from a conversation.
-    Returns a response based on the closest match based on in known conversations.
-    """
-    import random
+        if ratio > closest_ratio:
+            if responces_exist:
+                closest_ratio = ratio
+                closest_statement = statement
 
-    threshold = 90
-    closest_response, closest_ratio = get_closest(text, log_directory, threshold)
+        # If equal ratios, choose the one that has the greatest number of occurances
+        elif ratio == closest_ratio:
+            if responces_exist:
+                if statement["occurrences"] > closest_statement["occurrences"]:
+                    closest_statement = statement
 
-    if not closest_response:
-        from chatterbot.conversation import Statement
-        default = Statement("Error", "No possible replies could be determined.")
-        return [default]
+    # Select all statements which have been used to respond to this object
+    possible_responses = statements.find(
+        {"in_response_to": ObjectId(closest_statement["_id"])}
```

Comment:
Is there a reason why you are casting this to an `ObjectId`? I'm pretty sure it's not needed, and you aren't casting it in other places.